### ISSUE TITLE
Snackbar above BottomNavigation

### ISSUE THREAD

**Issue by Ben-Baert** [tingyik90/snackprogressbar#16]
How would I put the snackbar _above_ the bottom layout bar? 

the `viewsToMove`method would move the bottom bar, however, I want to snackbar to be above the bottom navigation bar.

The standard approach (with standard snackbars) is to set a margin on the bottom, but this doesn't seem to be possible with snackprogressbar.

Any advice?

**Comment by tingyik90** [comment]
I haven't consider that since I didn't use bottom navigation in my app. Have you tried [this](https://stackoverflow.com/a/36333022/3584439)?

The native behaviour of coordinate layout should work with snackbar by default. If it is really not working, I can consider exposing the snackbar view in this library.

**Comment by Ben-Baert** [comment]
Right now, the snackbar is just behind the bottom navigation view. I had a look at that SO thread before, the first solution is not really acceptable (the one with the many nested layouts) for performance reasons. I wanted to try the second one, but the view is not exposed, so I couldn't try that.

**Comment by tingyik90** [comment]
May I know which version you are using now? v5.0?

**Comment by Ben-Baert** [comment]
Yep, 5.0, as I've migrated everything to AndroidX recently.

**Comment by tingyik90** [comment]
Due to the way this library tackles the queuing issue, the snackbar is not drawn until it is its turn to be shown. I'll have to implement a callback for when the layout is inflated, in order to expose the view. 

Are you willing to help me test the snapshot of this library once updated? I should be able to get it done this weekend.

**Comment by Ben-Baert** [comment]
Sure!

It'd be useful to set a bottom margin directly on the manager that applies to all the snackbars, but any approach that works is fine.

**Comment by tingyik90** [comment]
Hi, I just updated the library. Please use `master-SNAPSHOT` instead of `v5.0`. I'm not going to add options to set the margin directly, since this is too specific. I'll go with a more generic approach. To get the view of snackbar, implement 

    .setOnDisplayListener(new SnackProgressBarManager.OnDisplayListener() {
        @Override
        public void onLayoutInflated(view: View, snackProgressBar: SnackProgressBar, onDisplayId: Int) {
            // Do your stuff
        }
    })

Let me know if it works.

**Comment by Ben-Baert** [comment]
Okay, sorry for the late reply. I'll try it today and let you know if it works!

Thanks a lot by the way!

**Comment by Ben-Baert** [comment]
This seems to work well!

For reference, the way I made this work:

```
SnackProgressBarManager.setOnDisplayListener(new SnackProgressBarManager.OnDisplayListener() {
                @Override
                public void onLayoutInflated(View view, SnackProgressBar snackProgressBar, int i) {
                    CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams)
                            view.getLayoutParams();
                    params.setMargins(0, 0, 0, bottomNavigationView.getItemHeight());
                    view.setLayoutParams(params);
                }
```

**Comment by tingyik90** [comment]
Thanks! Have you tested with `viewsToMove` in action as well? Are they animating the correct distance?

**Comment by Ben-Baert** [comment]
I'm going to play around with that a bit tonight and let you know.

**Comment by tingyik90** [comment]
How was it? Is the animation correct?

**Comment by Ben-Baert** [comment]
Sorry for the late reply!

Yes, the animation seems to work correctly indeed, at least in the scenarios I tested.

**Comment by tingyik90** [comment]
@Ben-Baert, this is now implemented officially in v6.1 with even better support. `SnackProgressBarLayout` is now public.
